### ISSUE TITLE
Datetime column breaking insertions into database

### ISSUE THREAD

**Issue by Nightyly** [rbock/sqlpp11#547]
Hi,
While trying to insert into a database, I noticed that when I add a Datetime column to the database and update the squema generated by ddl2cpp, and not changing the code at all, the insertion fails.

```cpp
auto db = sqlpp::mysql::connection(config);
auto t = oxim::Test{};
db(insert_into(t).set(t.userId = 1, t.heartrate = "[1, 2, 3]", t.spo2 = "[3, 4, 5]", t.heartrateprecision = 1.1, t.spo2Precision = 2.2));
```
This was the code for both cases.
Before adding the Datetime column, the ddl was the following:
```sql
-- --------------------------------------------------------
-- Host:                         127.0.0.1
-- Versión del servidor:         11.2.2-MariaDB - mariadb.org binary distribution
-- SO del servidor:              Win64
-- HeidiSQL Versión:             12.3.0.6589
-- --------------------------------------------------------

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET NAMES utf8 */;
/*!50503 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;


-- Volcando estructura de base de datos para oxim
CREATE DATABASE IF NOT EXISTS `oxim` /*!40100 DEFAULT CHARACTER SET latin1 COLLATE latin1_swedish_ci */;
USE `oxim`;

-- Volcando estructura para tabla oxim.test
CREATE TABLE IF NOT EXISTS `test` (
  `user_id` int(11) DEFAULT NULL,
  `heartrate` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL CHECK (json_valid(`heartrate`)),
  `spo2` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL CHECK (json_valid(`spo2`)),
  `heartrateprecision` double DEFAULT NULL,
  `spo2precision` double DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci ROW_FORMAT=DYNAMIC;

-- La exportación de datos fue deseleccionada.

/*!40103 SET TIME_ZONE=IFNULL(@OLD_TIME_ZONE, 'system') */;
/*!40101 SET SQL_MODE=IFNULL(@OLD_SQL_MODE, '') */;
/*!40014 SET FOREIGN_KEY_CHECKS=IFNULL(@OLD_FOREIGN_KEY_CHECKS, 1) */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40111 SET SQL_NOTES=IFNULL(@OLD_SQL_NOTES, 1) */;
```
And everything worked as expected.
After adding the Datetime column, the ddl was
```sql
-- --------------------------------------------------------
-- Host:                         127.0.0.1
-- Versión del servidor:         11.2.2-MariaDB - mariadb.org binary distribution
-- SO del servidor:              Win64
-- HeidiSQL Versión:             12.3.0.6589
-- --------------------------------------------------------

/*!40101 SET @OLD_CHARACTER_SET_CLIENT=@@CHARACTER_SET_CLIENT */;
/*!40101 SET NAMES utf8 */;
/*!50503 SET NAMES utf8mb4 */;
/*!40103 SET @OLD_TIME_ZONE=@@TIME_ZONE */;
/*!40103 SET TIME_ZONE='+00:00' */;
/*!40014 SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0 */;
/*!40101 SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='NO_AUTO_VALUE_ON_ZERO' */;
/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;


-- Volcando estructura de base de datos para oxim
CREATE DATABASE IF NOT EXISTS `oxim` /*!40100 DEFAULT CHARACTER SET latin1 COLLATE latin1_swedish_ci */;
USE `oxim`;

-- Volcando estructura para tabla oxim.test
CREATE TABLE IF NOT EXISTS `test` (
  `user_id` int(11) DEFAULT NULL,
  `heartrate` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL CHECK (json_valid(`heartrate`)),
  `spo2` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin DEFAULT NULL CHECK (json_valid(`spo2`)),
  `heartrateprecision` double DEFAULT NULL,
  `spo2precision` double DEFAULT NULL,
  `received` datetime DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1 COLLATE=latin1_swedish_ci ROW_FORMAT=DYNAMIC;

-- La exportación de datos fue deseleccionada.

/*!40103 SET TIME_ZONE=IFNULL(@OLD_TIME_ZONE, 'system') */;
/*!40101 SET SQL_MODE=IFNULL(@OLD_SQL_MODE, '') */;
/*!40014 SET FOREIGN_KEY_CHECKS=IFNULL(@OLD_FOREIGN_KEY_CHECKS, 1) */;
/*!40101 SET CHARACTER_SET_CLIENT=@OLD_CHARACTER_SET_CLIENT */;
/*!40111 SET SQL_NOTES=IFNULL(@OLD_SQL_NOTES, 1) */;
```
After updating the schema with ddl2cpp, and not changing the source code at all, the program compiled, but it crashes. The call stack is the following
```cpp
libstdc++-6.dll![Unknown/Just-In-Time compiled code] (Unknown Source:0)
sqlpp::text_operand::text_operand(sqlpp::text_operand * const this) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\data_types\text\operand.h:65)
sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>::assignment_t(sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand> * const this) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\assignment.h:52)
std::_Head_base<2ull, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, false>::_Head_base(std::_Head_base<2, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, false> * const this) (d:\msys64\ucrt64\include\c++\13.2.0\tuple:196)
std::_Tuple_impl<2ull, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >::_Tuple_impl(std::_Tuple_impl<2, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > * const this) (d:\msys64\ucrt64\include\c++\13.2.0\tuple:296)
std::_Tuple_impl<1ull, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >::_Tuple_impl(std::_Tuple_impl<1, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > * const this) (d:\msys64\ucrt64\include\c++\13.2.0\tuple:296)
std::_Tuple_impl<0ull, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >::_Tuple_impl(std::_Tuple_impl<0, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > * const this) (d:\msys64\ucrt64\include\c++\13.2.0\tuple:296)
std::tuple<sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >::tuple(std::tuple<sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > * const this) (d:\msys64\ucrt64\include\c++\13.2.0\tuple:901)
sqlpp::insert_list_data_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >::insert_list_data_t(sqlpp::insert_list_data_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > * const this) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\insert_value_list.h:130)
sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > >::statement_t<sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t>, sqlpp::insert_list_data_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > >(sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::insert_list_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > > * const this, sqlpp::statement_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t> statement, sqlpp::insert_list_data_t<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > term) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\statement.h:214)
sqlpp::no_insert_value_list_t::_base_t<sqlpp::detail::statement_policies_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t> >::_set_impl<void, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >(const sqlpp::no_insert_value_list_t::_base_t<sqlpp::detail::statement_policies_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t> > * const this, std::tuple<sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> > assignments) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\insert_value_list.h:596)
sqlpp::no_insert_value_list_t::_base_t<sqlpp::detail::statement_policies_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t> >::set<sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::UserId>, sqlpp::integral_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrate>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2>, sqlpp::text_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Heartrateprecision>, sqlpp::floating_point_operand>, sqlpp::assignment_t<sqlpp::column_t<oxim::Test, oxim::Test_::Spo2Precision>, sqlpp::floating_point_operand> >(const sqlpp::no_insert_value_list_t::_base_t<sqlpp::detail::statement_policies_t<void, sqlpp::insert_t, sqlpp::into_t<void, oxim::Test>, sqlpp::no_insert_value_list_t> > * const this) (d:\Docs\C++\OXIM-server\build\_deps\sqlpp11-src\include\sqlpp11\insert_value_list.h:539)
main() (d:\Docs\C++\OXIM-server\source\main.cpp:44)
```
But the funny thing is, when I remove every column from the table except userId and received, update squema and change the code to
```cpp
auto db = sqlpp::mysql::connection(config);
auto t = oxim::Test{};
db(insert_into(t).set(t.userId = 1, t.received = std::chrono::system_clock::now()));
```
everything works fine again.
Am I stupid?
Any help would be appreciated.

**Comment by rbock** [comment]
Thanks! However, I was not able to reproduce the crash (using both gcc and clang on Linux as I don't have access to a Windows machine).

Observations and questions:\
* You are using both mysql and mariadb in your CMakefile. What's the purpose of that? Is it possible that you compile with one and execute with the other?
* The crash message starts with `Unknown/Just-In-Time compiled code`. I am unfamiliar with debugging on Windows. Is that normal? What does it mean?
* Is it possible that the standard library used at runtime does not match the one at compile time?
* Afaict from the crash message, the crash is happening in the statement construction. IOW, it should crash without being actually sent to the database. Does it?

```
#include <sqlpp11/sqlpp11.h>

#include "../include/dbschema.h"

int main(){
    auto t = oxim::Test{};
    const auto x = insert_into(t).set(t.userId = 1, t.heartrate = "[1, 2, 3]", t.spo2 = "[3, 4, 5]", t.heartratePrecision = 1.1, t.spo2Precision = 2.2);
}
```

* Do the tests that come with the sqlpp11 library work for you?
* I slightly modified your sample and added it to the library tests, see [bug.diff.zip](https://github.com/rbock/sqlpp11/files/13539852/bug.diff.zip). Does that work for you?

**Comment by MeanSquaredError** [comment]
In the latest source code `sqlpp11\data_types\text\operand.h:65` contains
the default copy-constructor of `sqlpp::text_operand` :
```
    text_operand(const text_operand&) = default;
```
So the crash happens when the program tries to copy-construct `sqlpp::text_operand`. This struct contains only one property of type `std::string`,
so if the backtrace is valid then the program crashes when trying to copy a `std::string`, so either the heap is damaged and crashes on `malloc` or the source string is damaged. Either way indeed it points to memory corruption.

The `libstdc++-6.dll![Unknown/Just-In-Time compiled code` message looks strange.
@Nightyly , what language and compiler are you using? Does your platform support a memory checker either natively or through an external program similar to Valgrind or BoundsChecker?

**Comment by Nightyly** [comment]
I'm using Msys2 ucrt64 for gcc, cmake and make, and it seems like -fsanitize=address only works with msys2's clang.
As for the memory checker, I have used Deleaker on the executable, but I'm honestly not sure what I'm looking at.
Deleaker says there are 251 memory leaks, 3 of which are known leaks, 2 which have a source file associated with them and the remaining 246 do not.
I have hidden the leaks that are not directly associated with the executable, and there where 33 remaining
[Output is here](https://pastebin.com/5Xu3juHa)
Some of the 33 leaks had the following shared stack trace
```
OXIM-server.exe!_ZN5sqlpp5mysql6detail11thread_initEv Line 70 + 0x15 bytes 00007ff7ff61f675	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 70	
OXIM-server.exe!_ZN5sqlpp5mysql6detail17execute_statementERSt10unique_ptrINS1_17connection_handleESt14default_deleteIS3_EERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE Line 75 + 0x1a bytes 00007ff7ff61f83a	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 75	
OXIM-server.exe!_ZN5sqlpp5mysql15connection_base11insert_implERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE Line 208 + 0x1f bytes 00007ff7ff61f27f	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 208	
OXIM-server.exe!_ZN5sqlpp5mysql15connection_base6insertINS_11statement_tIvJNS_8insert_tENS_6into_tIvN4oxim4TestEEENS_13insert_list_tIvJNS_12assignment_tINS_8column_tIS7_NS6_5Test_6UserIdEEENS_16integral_operandEEENSA_INSB_IS7_NSC_9HeartrateEEENS_6null_tEEENSA_INSB_IS7_NSC_4Spo2EEESJ_EENSA_INSB_IS7_NSC_18HeartratePrecisionEEENS_22floating_point_operandEEENSA_INSB_IS7_NSC_14Spo2_precisionEEESQ_EENSA_INSB_IS7_NSC_8ReceivedEEESJ_EEEEEEEEEEyRKT_ Line 328 + 0x69 bytes 00007ff7ff61f349	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 328	
OXIM-server.exe!_ZNK5sqlpp8insert_t17_result_methods_tINS_11statement_tIvJS0_NS_6into_tIvN4oxim4TestEEENS_13insert_list_tIvJNS_12assignment_tINS_8column_tIS5_NS4_5Test_6UserIdEEENS_16integral_operandEEENS8_INS9_IS5_NSA_9HeartrateEEENS_6null_tEEENS8_INS9_IS5_NSA_4Spo2EEESH_EENS8_INS9_IS5_NSA_18HeartratePrecisionEEENS_22floating_point_operandEEENS8_INS9_IS5_NSA_14Spo2_precisionEEESO_EENS8_INS9_IS5_NSA_8ReceivedEEESH_EEEEEEEEE4_runINS_5mysql15connection_baseEEEDTcldtfp_6insertclptfpTL_ZNKSY_14_get_statementEvEEEERT_ Line 72 + 0x26 bytes 00007ff7ff624746	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\insert.h, line 72	
OXIM-server.exe!_ZNK5sqlpp11statement_tIvJNS_8insert_tENS_6into_tIvN4oxim4TestEEENS_13insert_list_tIvJNS_12assignment_tINS_8column_tIS4_NS3_5Test_6UserIdEEENS_16integral_operandEEENS7_INS8_IS4_NS9_9HeartrateEEENS_6null_tEEENS7_INS8_IS4_NS9_4Spo2EEESG_EENS7_INS8_IS4_NS9_18HeartratePrecisionEEENS_22floating_point_operandEEENS7_INS8_IS4_NS9_14Spo2_precisionEEESN_EENS7_INS8_IS4_NS9_8ReceivedEEESG_EEEEEEE4_runINS_5mysql15connection_baseEEEDTcldtclL_ZSt7declvalINS1_17_result_methods_tISW_EEEDTcl9__declvalIT_ELi0EEEvEE4_runfp_EERS13_ Line 243 + 0x20 bytes 00007ff7ff624280	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\statement.h, line 243	
OXIM-server.exe!_ZN5sqlpp5mysql15connection_base4_runINS_11statement_tIvJNS_8insert_tENS_6into_tIvN4oxim4TestEEENS_13insert_list_tIvJNS_12assignment_tINS_8column_tIS7_NS6_5Test_6UserIdEEENS_16integral_operandEEENSA_INSB_IS7_NSC_9HeartrateEEENS_6null_tEEENSA_INSB_IS7_NSC_4Spo2EEESJ_EENSA_INSB_IS7_NSC_18HeartratePrecisionEEENS_22floating_point_operandEEENSA_INSB_IS7_NSC_14Spo2_precisionEEESQ_EENSA_INSB_IS7_NSC_8ReceivedEEESJ_EEEEEEEEEEDTcldtfp_4_rundefpTEERKT_NS_12consistent_tE Line 419 + 0x1b bytes 00007ff7ff61f2cb	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 419	
OXIM-server.exe!_ZN5sqlpp5mysql15connection_baseclINS_11statement_tIvJNS_8insert_tENS_6into_tIvN4oxim4TestEEENS_13insert_list_tIvJNS_12assignment_tINS_8column_tIS7_NS6_5Test_6UserIdEEENS_16integral_operandEEENSA_INSB_IS7_NSC_9HeartrateEEENS_6null_tEEENSA_INSB_IS7_NSC_4Spo2EEESJ_EENSA_INSB_IS7_NSC_18HeartratePrecisionEEENS_22floating_point_operandEEENSA_INSB_IS7_NSC_14Spo2_precisionEEESQ_EENSA_INSB_IS7_NSC_8ReceivedEEESJ_EEEEEEEEEEDTclptfpT4_runfp_tlNS_9run_checkINS0_9context_tET_vE4typeEEEERKS12_ Line 428 + 0x1b bytes 00007ff7ff61f44b	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\connection.h, line 428	
OXIM-server.exe!main Line 19 + 0x22a bytes 00007ff7ff601685	D:\Docs\C++\OXIM-server_reprex\source\main.cpp, line 19	
OXIM-server.exe!mainCRTStartup Line 188 + 0x11 bytes 00007ff7ff601401	C:\M\B\src\mingw-w64\mingw-w64-crt\crt\crtexe.c, line 188	
```
and
```
OXIM-server.exe!_ZN5sqlpp5mysql6detail17connection_handleC1ERKSt10shared_ptrIKNS0_17connection_configEE Line 75 + 0x31 bytes 00007ff7ff61f701	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\mysql\detail\connection_handle.h, line 75	
OXIM-server.exe!_ZSt11make_uniqueIN5sqlpp5mysql6detail17connection_handleEJRKSt10shared_ptrIKNS1_17connection_configEEEENSt8__detail9_MakeUniqIT_E15__single_objectEDpOT0_ Line 1070 + 0x39 bytes 00007ff7ff6c31f9	D:\msys64\ucrt64\include\c++\13.2.0\bits\unique_ptr.h, line 1070	
OXIM-server.exe!_ZN5sqlpp6compat11make_uniqueINS_5mysql6detail17connection_handleEJRKSt10shared_ptrIKNS2_17connection_configEEEEESt10unique_ptrIT_St14default_deleteISC_EEDpOT0_ Line 45 + 0x26 bytes 00007ff7ff620166	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\compat\make_unique.h, line 45	
OXIM-server.exe!_ZN5sqlpp17normal_connectionINS_5mysql15connection_baseEEC1ERKSt10shared_ptrIKNS1_17connection_configEE Line 75 + 0x22 bytes 00007ff7ff61e8d2	D:\Docs\C++\OXIM-server_reprex\build\_deps\sqlpp11-src\include\sqlpp11\connection.h, line 75	
OXIM-server.exe!main Line 15 + 0xea bytes 00007ff7ff601545	D:\Docs\C++\OXIM-server_reprex\source\main.cpp, line 15	
OXIM-server.exe!mainCRTStartup Line 188 + 0x11 bytes 00007ff7ff601401	C:\M\B\src\mingw-w64\mingw-w64-crt\crt\crtexe.c, line 188	
```
main.cpp
```cpp
#include <cstdint>

#include <sqlpp11/sqlpp11.h>
#include <sqlpp11/mysql/mysql.h>

#include "../include/dbschema.h"

int main(){
    auto config = std::make_shared<sqlpp::mysql::connection_config>();
    config -> database = "oxim";
    config -> user = "root";
    config -> password = "root";
    config -> host = "localhost";
    config -> debug = false;
    auto db = sqlpp::mysql::connection(config);
    auto t = oxim::Test{};
    try{
        const auto x = insert_into(t).set(t.userId = 1, t.heartrate = sqlpp::null, t.spo2 = sqlpp::null, t.heartratePrecision = 1.1, t.spo2_precision = 2.2, t.received = sqlpp::null);
        db(x);
    }
    catch(...){}
    std::cout << "a";
    return 0;
}
```
From what I can see, there where no memory leaks started from line 18 of main.cpp, which is the point at which the program crashes

**Comment by MeanSquaredError** [comment]
@Nightyly 
I suggested using a memory checker because I suspected that there is memory corruption (damaged heap, stack buffer overrun, etc.). I think that the memory leaks are unrelated to the problem.

Exit code `0xc0000409` is called `STATUS_STACK_BUFFER_OVERRUN`, but the name is sort of obsolete now. Exit code `0xc0000409` means that the program encountered a fatal program and did a fast-fail exit without a proper cleanup. It could be memory corruption, a function being called with an invalid argument (e.g. NULL pointer), other problem where the program called `abort()` or `std::terminate()`, etc.
Raymond Chen describes it in slightly more detail here
https://devblogs.microsoft.com/oldnewthing/20190108-00/?p=100655

In your case the fast-fail happens in `libstdc++-6.dll` most likely during the copy-construction a `std::string`. So it could be memory corruption (e.g. heap corruption), or some of the memory-handling functions libstdc++ is being called with an invalid pointer. Or it could be some other critical problem detected inside libstdc++.

I suggest installing the debug symbols for libstdc++-6.dll and the sources for libstdc++-6, then compiling your minimal reproducible example with debug symbols, then running it in gdb and seeing where exactly the problem happens in libstdc++-6. The source code at and around the place where libstdc++-6 does the fast-fail should hint at what exactly goes wrong.
